packages/adblocker/src/lists.ts

```diff
@@ -189,38 +202,112 @@ export function parseFilters(
       const filter = NetworkFilter.parse(line, config.debug);
       if (filter !== null) {
         networkFilters.push(filter);
+        if (preprocessorStack.length) {
+          filterStack.push(filter);
+        }
       }
     } else if (filterType === FilterType.COSMETIC && config.loadCosmeticFilters === true) {
       const filter = CosmeticFilter.parse(line, config.debug);
       if (filter !== null) {
         if (config.loadGenericCosmeticsFilters === true || filter.isGenericHide() === false) {
           cosmeticFilters.push(filter);
+          if (preprocessorStack.length) {
+            filterStack.push(filter);
+          }
+        }
+      }
+    } else if (config.loadPreprocessors) {
+      const preprocessorType = detectPreprocessor(line);
+
+      if (preprocessorType === PreprocessorTypes.BEGIF) {
+        preprocessorStack.push({
+          start: filterStack.length,
+          condition: Preprocessor.getCondition(line),
+        });
+      } else if (preprocessorType !== PreprocessorTypes.INVALID && preprocessorStack.length > 0) {
```

Comment:
I would rather explicitly list which cases are accepted here for clarity:
```suggestion
      } else if ((preprocessorType === PreprocessorTypes.ENDIF || preprocessorType ===  && PreprocessorTypes.ELSE) && preprocessorStack.length > 0) {
```